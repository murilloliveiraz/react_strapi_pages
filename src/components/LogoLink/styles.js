import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${() => css`
    text-decoration: none;
    color: inherit;
    display: flex;
    align-items: center;

    > img {
      max-height: 3rem;
    }
  `}
`;
