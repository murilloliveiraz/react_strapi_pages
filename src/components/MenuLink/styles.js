import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    padding: ${theme.spacings.small};
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.7rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 200ms;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
