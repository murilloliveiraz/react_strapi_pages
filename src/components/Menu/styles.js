import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    border-bottom: ${theme.colors.mediumGray};

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      /* visibility: hidden;
      opacity: 0; */

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      /* padding: ${theme.spacings.xxlarge}; */
    }
  `}
`;

export const Button = styled.div`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme};
  `}
`;
