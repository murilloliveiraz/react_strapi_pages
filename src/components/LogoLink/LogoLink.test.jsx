import { screen } from '@testing-library/react';
import { LogoLink } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<LogoLink/>', () => {
  it('shoould render TextLogo', () => {
    renderTheme(<LogoLink link="#target" text="Olá Mundo" />);
    expect(screen.getByRole('link', { name: 'Olá Mundo' })).toHaveAttribute(
      'href',
      '#target',
    );
    expect(screen.getByRole('link', { name: 'Olá Mundo' })).toBeInTheDocument();
  });

  it('shoould render ImageLogo', () => {
    renderTheme(
      <LogoLink link="#target" text="Olá Mundo" srcImg="image.jpg" />,
    );
    expect(screen.getByAltText('Olá Mundo')).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });

  it('shoould match snapshot', () => {
    renderTheme(
      <LogoLink link="#target" text="Olá Mundo" srcImg="image.jpg" />,
    );
    expect(screen.getByAltText('Olá Mundo')).toMatchSnapshot();
  });
});
