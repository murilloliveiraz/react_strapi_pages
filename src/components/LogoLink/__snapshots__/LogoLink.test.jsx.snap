// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink/> shoould match snapshot 1`] = `
<img
  alt="OlÃ¡ Mundo"
  src="image.jpg"
/>
`;
